src/reducers/__tests__/existingThoughtChange.js

```diff
@@ -10,27 +10,176 @@ it('edit a thought', () => {
     newThought({ value: 'b' }),
     setCursor({ thoughtsRanked: [{ value: 'a', rank: 0 }] }),
     existingThoughtChange({
-      newValue: 'c',
+      newValue: 'aa',
       oldValue: 'a',
       context: [ROOT_TOKEN],
-      thoughtsRanked: [ { value: 'a', rank: 0 }, { value: 'b', rank: 1 } ]
+      thoughtsRanked: [{ value: 'a', rank: 0 }]
     })
   ]
+  // run steps through reducer flow and export as plaintext for readable test
+  const stateNew = reducerFlow(steps)(initialState())
+  const exported = exportContext(stateNew, [ROOT_TOKEN], 'text/plaintext')
+
+  expect(exported).toBe(`- ${ROOT_TOKEN}
+  - aa
+  - b`)
+
+  // aa should exist in ROOT context
+  expect(getContexts(stateNew, 'aa'))
+    .toMatchObject([{
+      context: [ROOT_TOKEN]
+    }])
+  expect(getThoughts(stateNew, [ROOT_TOKEN]))
+    .toMatchObject([{ value: 'b', rank: 1 }, { value: 'aa', rank: 0 }])
+
+  expect(stateNew.cursor)
```

Comment:
Cursor does not update upon editing the active thought.

Reply:
@raineorshine The state.cursor in this case returns the original path, before edit, i.e `[ { value: 'a', rank: 0 } ]`

Reply:
Good catch. It appears that `existingThoughtChange` does not update the `cursor` at all. This will be fixed separately.